HttpClientCases.cs

```diff
@@ -0,0 +1,186 @@
+﻿using HelloWorld.Models;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HelloWorld
+{
+    public class HttpClientCases
+    {
+        string BaseUrl = "https://restful-booker.herokuapp.com/";
+
+        public async Task<string> GetToken()
+        {
+            using (var client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                var data = new Dictionary<string, string>
+                    {
+                        {"username", "admin"},
+                        {"password", "password123"}
+                    };
+                var response = await client.PostAsync(BaseUrl + "auth", new FormUrlEncodedContent(data));
+                response.EnsureSuccessStatusCode();
+                var token = await response.Content.ReadAsStringAsync();
+                var tokenRaw = token.DeserializeJson<AuthToken>();
+                return tokenRaw.Token;
+            }
+        }
+
+        public async Task GetBookingIDs()
+        {
+            using (var client = new HttpClient())
+            {
+                try
```

Comment:
remember we've discussed the purpose of **using** statement? there's no need to wrap things into try-catch within the using:
![image](https://user-images.githubusercontent.com/3397575/210329367-0af26c87-274c-44fa-ad8b-c834bda35441.png)